activitysim/abm/models/vehicle_type_choice.py

```diff
@@ -0,0 +1,444 @@
+    missing_alts = choosers.loc[choosers._merge == 'left_only', ['vehicle_type']]
+    assert len(missing_alts) == 0, \
+        f"missing probabilities for alternatives:\n {missing_alts}"
+    choosers.drop(columns='_merge', inplace=True)
+    del choosers['vehicle_type']
+
+    # probs should sum to 1 with residual probs resulting in choice of 'fail'
+    chooser_probs = choosers.div(choosers.sum(axis=1), axis=0).fillna(0)
+    chooser_probs['fail'] = 1 - chooser_probs.sum(axis=1).clip(0, 1)
+
+    # make probabilistic choices
+    prob_choices, rands = logit.make_choices(chooser_probs, trace_label=trace_label, trace_choosers=choosers)
+
+    # convert alt choice index to vehicle type attribute
+    prob_choices = chooser_probs.columns[prob_choices.values].to_series(index=prob_choices.index)
+    failed = (prob_choices == chooser_probs.columns.get_loc('fail'))
+    prob_choices = prob_choices.where(~failed, "NOT CHOSEN")
+
+    # add new attribute to logit choice vehicle types
+    choices['vehicle_type'] = choices['vehicle_type'] + '_' + prob_choices
+
+    return choices
+
+
+def annotate_vehicle_type_choice(model_settings, trace_label):
+    """
+    Contains the calls to optionally annotate the persons, households, or
+    vehicles tables.  Adds columns to the corresponding output tables.
+
+    Parameters
+    ----------
+    model_settings : dict
+    trace_label : str
+
+    Returns
+    -------
+    """
+    # - annotate households table
+    if model_settings.get('annotate_households'):
+        households = inject.get_table('households').to_frame()
+        expressions.assign_columns(
+            df=households,
+            model_settings=model_settings.get('annotate_households'),
+            trace_label=tracing.extend_trace_label(trace_label, 'annotate_households'))
+        pipeline.replace_table("households", households)
+
+    # - annotate persons table
+    if model_settings.get('annotate_persons'):
+        persons = inject.get_table('persons').to_frame()
+        expressions.assign_columns(
+            df=persons,
+            model_settings=model_settings.get('annotate_persons'),
+            trace_label=tracing.extend_trace_label(trace_label, 'annotate_persons'))
+        pipeline.replace_table("persons", persons)
+
+    # - annotate vehicles table
+    if model_settings.get('annotate_vehicles'):
+        vehicles = inject.get_table('vehicles').to_frame()
+        expressions.assign_columns(
+            df=vehicles,
+            model_settings=model_settings.get('annotate_vehicles'),
+            trace_label=tracing.extend_trace_label(trace_label, 'annotate_vehicles'))
+        pipeline.replace_table("vehicles", vehicles)
+
+
+def iterate_vehicle_type_choice(
+        vehicles_merged,
+        model_settings,
+        model_spec,
+        locals_dict,
+        estimator,
+        chunk_size,
+        trace_label):
+    """
+    Iterates through household vehicle numbers and selects a vehicle type of
```

Comment:
Suggest running this thru [pydocstyle](http://www.pydocstyle.org/en/stable/) for general suggestions on how the docstrings are formed. For example: the summary description of a function should be fit on a single line and end with punctuation.

Reply:
done.  Cleaned up docstrings.